openstack-rabbit-consumer/rabbit_consumer/openstack_api.py

```diff
@@ -92,10 +92,14 @@ def get_image(vm_data: VmData) -> Optional[Image]:
     Gets the image name from Openstack for the virtual machine.
     """
     server = get_server_details(vm_data)
-    uuid = server.image.id
-    if not uuid:
+
+    if not (server.image and server.image.id):
+        # User has booted from a volume, so we have an image without an ID
+        # or no image at all...
         return None
```

Comment:
I prefer this because the if statement was slightly confusing 
```suggestion
    image = getattr(server, "image", None)
    # User has booted from a volume, so we have an image without an ID
    if image is not None:
        return getattr(image, "id", None)
    # or no image at all...
    return None
```
if server_image:
    return getattr(server_image, "id", None)
return None

Reply:
Hi @anish-mudaraddi , 
I am not sure your proposed logic is correct. In your case, you may return the image ID, while I understand the expected output of this function is to return, when possible, an entire image object.

Reply:
But it is true that @DavidFair has tried to be smart with his if statement, using the De Morgan's Law

    Not(X AND Y) == ( Not(X) OR Not(Y) )

Maybe, for those who are not as smart as David, it would be more clear just to do

if ( not server.image or not server.image.id):
    return None

Reply:
@jose-caballero - yes you're right

bottom line is I'd prefer to break the if statement into two parts so its more clear
 
```suggestion
    image = getattr(server, "image", None)
    if not image:
        return None 
    # User has booted from a volume, so we have an image without an ID
    if not getattr(image, "id", None):
        return None
```

Reply:
Agreed, looking at what other people are using lets follow the Python's best practice of EAFP: https://stackoverflow.com/a/43491315